tests/cnf/core/network/policy/tests/multinetpolicy_bondoversriov.go

```diff
@@ -0,0 +1,729 @@
+		tNs1, err = namespace.NewBuilder(APIClient, testNs1).WithMultipleLabels(params.PrivilegedNSLabels).
+			WithLabel("ns", "ns1").Create()
+		Expect(err).ToNot(HaveOccurred(), "Failed to create test namespace")
+		tNs2, err = namespace.NewBuilder(APIClient, testNs2).WithMultipleLabels(params.PrivilegedNSLabels).
+			WithLabel("ns", "ns2").Create()
+		Expect(err).ToNot(HaveOccurred(), "Failed to create test namespace")
+
+		By("Deploy Test Resources: 2 Sriov Policies")
+		_, err = sriov.NewPolicyBuilder(APIClient,
+			"nicpf1", NetConfig.SriovOperatorNamespace, nicPf1, 5,
+			[]string{sriovInterfacesUnderTest[0]}, NetConfig.WorkerLabelMap).
+			WithDevType("netdevice").
+			Create()
+		Expect(err).ToNot(HaveOccurred(), "Failed to create test policy")
+
+		_, err = sriov.NewPolicyBuilder(APIClient,
+			"nicpf2", NetConfig.SriovOperatorNamespace, nicPf2, 5,
+			[]string{sriovInterfacesUnderTest[1]}, NetConfig.WorkerLabelMap).
+			WithDevType("netdevice").
+			Create()
+		Expect(err).ToNot(HaveOccurred(), "Failed to create test policy")
+
+		By("Deploy Test Resources: 4 Sriov Networks")
+		defineAndCreateSriovNetwork(ns1+nicPf1, nicPf1, testNs1)
+		defineAndCreateSriovNetwork(ns1+nicPf2, nicPf2, testNs1)
+		defineAndCreateSriovNetwork(ns2+nicPf1, nicPf1, testNs2)
+		defineAndCreateSriovNetwork(ns2+nicPf2, nicPf2, testNs2)
+
+		err = netenv.WaitForSriovAndMCPStable(APIClient, tsparams.MCOWaitTimeout, 10*time.Second,
+			NetConfig.CnfMcpLabel, NetConfig.SriovOperatorNamespace)
+		Expect(err).ToNot(HaveOccurred(), "Sriov and MCP are not stable")
+
+		By("Deploy Test Resources: 2 NADs for bond CNI")
+		defineAndCreateBondNAD("bond", testNs1)
+		defineAndCreateBondNAD("bond", testNs2)
+
+		By("Deploy Test Resources: 5 Pods")
+		testPod1 = defineAndCreatePodWithBondIf(pod1, testNs1, ns1+nicPf1, ns1+nicPf2, testData)
+		testPod2 = defineAndCreatePodWithBondIf(pod2, testNs1, ns1+nicPf1, ns1+nicPf2, testData)
+		testPod3 = defineAndCreatePodWithBondIf(pod3, testNs1, ns1+nicPf1, ns1+nicPf2, testData)
+		testPod4 = defineAndCreatePodWithBondIf(pod4, testNs2, ns2+nicPf1, ns2+nicPf2, testData)
+		testPod5 = defineAndCreatePodWithBondIf(pod5, testNs2, ns2+nicPf1, ns2+nicPf2, testData)
+
+		// Expected Result 1's and 0's represent Pass and Fail respectively when converted to Binary.
+		// Since we have 3 ports per destination pod, Octal notation is intuitive for readability.
+		// Example: expected Result (Octal) 0o70 --> (Binary) 0b111000 represents as below for the destination pod.
+		//
+		//   Pass     Pass     Pass     Fail     Fail     Fail
+		//     1       1        1        0         0       0
+		// tcp:5001 tcp:5002 udp:5003 tcp:5001 tcp:5002 udp:5003
+		// --IPv4-- --IPv4-- --IPv4-- --IPv6-- --IPv6-- --IPv6--
+		// -------------------Destination Pod-------------------
+		//
+		// Appropriate octal number is passed for each test.
+		By("Check egress traffic from pod1 to other 4 pods. All ports should be accessible")
+		defineNmapValidPaths(
```

Comment:
what if we change the function from accepting arrays of source pods, destination pods and expectedResults to a function that has 1 source pods, 1 destination pod, 1 expected result (encoded as you prefer)?
It would be a little more verbose in the call site, but we would gain in readability. WDYT?

Reply:
Currently, we are calling this function 2 times per test case. If we call it once per source pod--destination pod, we need to call 6 more times per test case and repeat that for all 13 test cases.
So, I just wanted to keep the code shorter instead